packages/editor/src/tools/ivy-change-bounds-manager.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  ChangeBoundsManager,
+  FeedbackEmitter,
+  GModelElement,
+  SelectionService,
+  SetUIExtensionVisibilityAction,
+  TrackedMove,
+  TrackedResize
+} from '@eclipse-glsp/client';
+import { inject, injectable } from 'inversify';
+import { QuickActionUI } from '../ui-tools/quick-action/quick-action-ui';
+import { ShowNegativeAreaAction } from './negative-area/model';
+
+@injectable()
+export class IvyChangeBoundsManager extends ChangeBoundsManager {
+  @inject(SelectionService) protected selectionService: SelectionService;
+
+  addMoveFeedback(feedback: FeedbackEmitter, trackedMove: TrackedMove, ctx?: GModelElement, event?: MouseEvent): FeedbackEmitter {
+    super.addMoveFeedback(feedback, trackedMove, ctx, event);
+    this.hideQuickUIFeedback(feedback);
+    this.showNegativeArea(feedback);
+    return feedback;
+  }
+
+  addResizeFeedback(feedback: FeedbackEmitter, resize: TrackedResize, ctx?: GModelElement, event?: MouseEvent): FeedbackEmitter {
+    super.addResizeFeedback(feedback, resize, ctx, event);
+    this.hideQuickUIFeedback(feedback);
+    this.showNegativeArea(feedback);
+    return feedback;
+  }
+
+  protected hideQuickUIFeedback(feedback: FeedbackEmitter): void {
+    feedback.add(
+      SetUIExtensionVisibilityAction.create({ extensionId: QuickActionUI.ID, visible: false }),
+      SetUIExtensionVisibilityAction.create({
+        extensionId: QuickActionUI.ID,
+        visible: true,
+        contextElementsId: [...this.selectionService.getSelectedElementIDs()]
+      })
```

Comment:
is the second to show the quickaction UI again after the move/resize

Reply:
Exactly! We introduced this feedback emitter helper object and now when you add a feedback action you can already specify how that action should be cleaned up as soon as the feedback gets disposed. So in this case, during move or resize we hide but as soon as we no longer move/resize and dispose that feedback we show the quickaction UI again. Same for the negative area: We show it during move/resize but hide it again after we are done.